component/azstorage/file_share.go

```diff
@@ -150,86 +154,259 @@ func (fs *FileShare) TestPipeline() error {
+	prop, err := fileURL.GetProperties(context.Background())
+
+	if err == nil { // file
+		attr = &internal.ObjAttr{
+			Path:   name, // We don't need to strip the prefixPath here since we pass the input name
+			Name:   filepath.Base(name),
+			Size:   prop.ContentLength(),
+			Mode:   0,
+			Mtime:  prop.LastModified(),
+			Atime:  prop.LastModified(),
+			Ctime:  prop.LastModified(),
+			Crtime: prop.LastModified(),
+			Flags:  internal.NewFileBitMap(),
+		}
+		parseMetadata(attr, prop.NewMetadata())
+		attr.Flags.Set(internal.PropFlagMetadataRetrieved)
+		attr.Flags.Set(internal.PropFlagModeDefault)
+
+		return attr, nil
+	} else { // directory
+		dirURL := fs.Share.NewDirectoryURL(filepath.Join(fs.Config.prefixPath, name))
+		log.Trace("FileShare::GetAttr : getting directoryyyyyyyyyyyyyy properties for %s, %s, %s", fs.Config.prefixPath, name, dirURL)
+		prop, err := dirURL.GetProperties(context.Background())
+
+		if err == nil {
+			attr = &internal.ObjAttr{
+				Path:   name,
+				Name:   filepath.Base(name),
+				Size:   4096,
+				Mode:   0,
+				Mtime:  prop.LastModified(),
+				Atime:  prop.LastModified(),
+				Ctime:  prop.LastModified(),
+				Crtime: prop.LastModified(),
+				Flags:  internal.NewDirBitMap(),
+			}
+			parseMetadata(attr, prop.NewMetadata())
+			attr.Flags.Set(internal.PropFlagMetadataRetrieved)
+			attr.Flags.Set(internal.PropFlagModeDefault)
+
+			return attr, nil
+		} else { // error
+			e := storeFileErrToErr(err)
+			if e == ErrFileNotFound {
+				return attr, syscall.ENOENT
+			} else {
+				log.Err("FileShare::GetAttr : Failed to get file/directory properties for %s (%s)", name, err.Error())
+				return attr, err
+			}
+		}
+
+	}
+
 }
 
-// Standard operations to be supported by any account type
+// List : Get a list of blobs matching the given prefix
+// This fetches the list using a marker so the caller code should handle marker logic
+// If count=0 - fetch max entries
 func (fs *FileShare) List(prefix string, marker *string, count int32) ([]*internal.ObjAttr, *string, error) {
-	return nil, nil, nil
+	log.Trace("FileShare::List : prefix %s, marker %s", prefix, func(marker *string) string {
+		if marker != nil {
+			return *marker
+		} else {
+			return ""
+		}
+	}(marker))
+
+	fileList := make([]*internal.ObjAttr, 0)
+
+	if count == 0 {
+		count = common.MaxDirListCount
+	}
+
+	listPath := filepath.Join(fs.Config.prefixPath, prefix)
+
+	// Get a result segment starting with the file indicated by the current Marker.
+	var listFile *azfile.ListFilesAndDirectoriesSegmentResponse
+	var err error
+	if listPath != "" {
+		listFile, err = fs.Share.NewDirectoryURL(listPath).ListFilesAndDirectoriesSegment(context.Background(), azfile.Marker{Val: marker},
+			azfile.ListFilesAndDirectoriesOptions{MaxResults: count})
+	} else {
+		if (prefix != "" && prefix[len(prefix)-1] == '/') || (prefix == "" && fs.Config.prefixPath != "") {
```

Comment:
is there a point to doing this check and changing listPath? I think you might be able to just use the NewDirectoryURL code above and get rid of this if else. NewDirectoryURL("") should just return whatever NewRootDirectoryURL does